src/player.cpp

```diff
@@ -5450,14 +5450,76 @@ void player::suffer()
         g->is_in_sunlight( pos() ) && one_turn_in( 1_minutes ) ) {
         // Umbrellas can keep the sun off the skin and sunglasses - off the eyes.
         if( !weapon.has_flag( "RAIN_PROTECT" ) ) {
-            add_msg_if_player( m_bad, _( "The sunlight is really irritating your skin." ) );
-            if( has_effect( effect_sleep ) && !has_effect( effect_narcosis ) ) {
-                wake_up();
+            //calculate total coverage of skin
+            body_part_set affected_bp { {
+                    bp_leg_l, bp_leg_r, bp_torso, bp_head, bp_mouth, bp_arm_l,
+                    bp_arm_r, bp_foot_l, bp_foot_r, bp_hand_l, bp_hand_r
+                }
+            };
+            //pecentage of "open skin" by body part
+            std::map<body_part, float> open_percent;
+            //initialize coverage
+            for( const body_part &bp : all_body_parts ) {
+                if( affected_bp.test( bp ) ) {
+                    open_percent[bp] = 1.0;
+                }
             }
-            if( one_turn_in( 1_minutes ) ) {
-                mod_pain( 1 );
-            } else {
-                focus_pool --;
+            //calculate coverage for every body part
+            for( const item &i : worn ) {
+                body_part_set covered = i.get_covered_body_parts();
+                for( const body_part &bp : all_body_parts )  {
+                    if( !affected_bp.test( bp ) || !covered.test( bp ) ) {
+                        continue;
+                    }
+                    //percent of "not covered skin"
+                    float p = 1.0 - i.get_coverage() / 100.0;
+                    open_percent[bp] = open_percent[bp] * p;
+                }
+            }
+
+            body_part max_affected_bp = num_bp;
+            float max_affected_bp_percent = 0;
+            int count_affected_bp = 0;
+            for( auto &it : open_percent ) {
+                const body_part &bp = it.first;
+                float &p = it.second;
+
+                if( p < 1.0 ) {
```

Comment:
I think it's supposed to be `p <= 0.0`? Otherwise, if my understanding of your code is correct, only bodyparts completely uncovered would be irritated.

I also feel this should be a less strict bound before the sunlight no longer irritates a body part, but I'm not sure how well an albino person is supposed to cover their body IRL to avoid sunlight irritation.

Reply:
Oh, I see in your OP that it's supposed to apply the abino effects below 99% coverage. In that case this should be `p <= 0.01` since you are using a fraction instead of a percentage here.

Reply:
>it's supposed to be `p <= 0.0`?

no, actually it is the check for 99% or more coverage:

>However, if all body parts coverage are 99% or better, then no albino effect is applied.

`p` starts from 1, means that part is 100% uncovered. Then this number is multiplied by `(1-item_coverage)`. So if the item covers 95%, it will be `p * 0.05 = 5%` uncovered now.
Parts having 1% or less uncovered are considered 100% covered.
Tested this formula with full survivor gear (100% coverage). Works fine, not more skin irritation at all.

>I also feel this should be a less strict bound before the sunlight no longer irritates a body part.

Agreed. But I also don't have scientific proofs for some exact numbers.
How about guessing that 95% covered is fine?

Reply:
Alternatively the calculation process can be changed so that for more coverage it will give less harmful effect, or do it less often? 
For ex:
0% coverage - full effect, once in a minute.
50% coverage - half effect, once in 2 minutes.
95% coverage - no effect.

And apply some approximation to it.

Reply:
>`p <= 0.01`

yes, fixed. Changed magic number to constant;

> p <= COVERAGE_LIMIT